virttest/utils_misc.py

```diff
@@ -3045,6 +3045,79 @@ def get_windows_drive_letters(session):
     return drive_letters
 
 
+class IozoneTest(object):
+    """
+    Run iozone test on guest
+    """
+    def __init__(self, session, params):
+        """
+        run setup of iozone test
+        """
+        self.params = params
+        self.session = session
+        self.timeout = float(params.get("disk_op_timeout", 240))
+        self.iozone_version = params.get("iozone_version")
+        if self.params.get("os_type") == "linux":
+            iozone_status = self.check_linux_iozone()
+            if not iozone_status:
+                self.setup_linux_iozone()
+
+    def setup_linux_iozone(self):
+        """
+        Download iozone and complie it
+        """
+        pre_cmd = self.params.get("pre_cmd") % (self.iozone_version,
+                                                self.iozone_version, self.iozone_version)
+        if pre_cmd:
+            self.session.cmd(pre_cmd, timeout=self.timeout)
+            if self.params.get("clean_iozone_tar"):
+                self.session.cmd(self.params["clean_iozone_tar"] % self.iozone_version,
+                                 timeout=self.timeout)
+
+    def check_linux_iozone(self):
+        """
+        check if iozone exist in linux guest
+        """
+        cmd = self.params.get("check_iozone_cmd") % self.iozone_version
+        output = self.session.cmd_output(cmd)
+        if "iozone -h" in output:
+            return True
+        else:
+            return False
+
+    def run_iozone_in_guest(self, disk_index, mountpoint):
+        """
+        Do iozone test in guest
+        """
+        iozone_test_cmd = self.params.get("iozone_test_cmd")
+        if self.params.get("os_type") == "windows":
+            format_windows_disk(self.session, disk_index, mountpoint)
+            test_cmd = iozone_test_cmd % (mountpoint, mountpoint)
+            test_cmd = set_winutils_letter(self.session, test_cmd)
+        elif self.params.get("os_type") == "linux":
+            test_cmd = iozone_test_cmd % mountpoint
+
+        status, output = self.session.cmd_status_output(test_cmd,
+                                                        timeout=self.timeout)
+        return status, output
+
+    def iozone_clean(self):
+        """
+        Clean old iozone packet after test in linux guest
+        """
+        if self.params.get("os_type") == "linux":
+            iozone_check = True
+            while iozone_check:
+                output = self.session.cmd_output("ps aux | grep iozone | grep -v grep")
+                if "iozone" not in output:
+                    iozone_check = False
+                    break
+                time.sleep(5)
+            cmd = "cd /home &&"
+            cmd += "rm -rf iozone*"
```

Comment:
does setup_linux_iozone uses /home to download ?, how does this path have iozone